### PR TITLE
[Java] CWE-078: Add JSch lib OS Command Injection sink

### PR DESCRIPTION
**Pull request by p0wn4j**
JSch is a pure Java implementation of SSH2. JSch allows you to connect to a sshd server and use port forwarding, X11 forwarding, file transfer, command execution, etc. 
If an attacker can control the command that is going to be executed on a sshd server, it can lead an attacker to Remote Code Execution on that server.

Code example:
```
JSch jsch = new JSch();
...
Session session = jsch.getSession(user, host, 22);
...
session.connect();
Channel channel = session.openChannel("exec");
((ChannelExec) channel).setCommand(controlledCommand);
channel.connect();
...
```

Found hits on these repos:
https://github.com/KingLai23/Puzzle
https://github.com/manuelacalvo/DATATourisme
https://github.com/jack775544/jTransfer

### CODE REVIEW

**Review by owen-mc — CHANGES_REQUESTED** [review]
You can simplify by making `setCommand` the sink. I can't think of any good reason why code would call `setCommand` apart from that they intend to call `connect` soon afterwards.

This shouldn't be a stand-alone query. It should be adding some sinks to `ExecTainted`, the existing command injection query. That query isn't very easy to extend at the moment, so I have [made a PR](https://github.com/github/codeql/pull/5746) to fix that.

File: java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjection.qhelp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+</p>
+</overview>
+
+<recommendation>
+<p>
+Including an user input in a JSch lib's OS command expression should be avoided.
+If this is not possible, then a strong input validation must be performed. 
+Some examples of an effective validation include:
+
+Validating against a whitelist of permitted values.
```
[the remark this answers is not in the document]
Comment (p0wn4j) [reply java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjection.qhelp:21]: Ok thanks :)
But in this PR https://github.com/github/codeql/pull/5746 classes are not from experemental queries, can I extend ExecTainted sink?

File: java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjectionSanitized.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        String[] badChars = new String[] {"^", "~" ," " , "&", "|", ";", "$", ">", "<", "`", "\\", ",", "!", "{", "}", "(", ")", "@", "%", "#", "%0A", "%0a", "\n", "\r\n"};
+
+        for (String badChar : badChars) {
+            if (untrusted.contains(badChar)) {
+                System.out.println("Invalid IP address");
+                return;
+            }    
+        }
+
+        //GOOD: Validate user the input.
```
[the remark this answers is not in the document]
Comment (p0wn4j) [reply java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjectionSanitized.java:15]: I think there maybe cases where just validation with regexp is not enough, that's why both of checks will make more secure.

File: java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjection.qhelp (hunk 1 above)
[the remark this answers is not in the document]
Comment (owen-mc) [reply java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjection.qhelp:21]: I think you can put the classes extending `ExecCallable` into a separate `qll` file, then make a copy of `ExecTainted.ql` which also imports that `qll` file.

**Comment by owen-mc** [comment]
@p0wn4j [My PR](https://github.com/github/codeql/pull/5746) has been merged now, so you can extend the `ExecTainted` query easily.

File: java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjectionSanitized.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (owen-mc) [reply java/ql/src/experimental/Security/CWE/CWE-094/JSchOSInjectionSanitized.java]: In that case I don't see the point in having the first check.

**Review by owen-mc — COMMENTED** [review]
That looks great. I've checked and the new version gets the same hits on LGTM as the old one. The other, more minor, comments from my previous review still stand.

**Comment by owen-mc** [comment]
@p0wn4j Do you intend to make a bug bountry submission for this contribution?

**Comment by p0wn4j** [comment]
Yes I want to make a bug bounty submission.

**Comment by owen-mc** [comment]
@p0wn4j Great, that addresses all of the review comments. When you make a bug bountry submission this will be ready to pass to the next stage.

**Comment by p0wn4j** [comment]
OK, thanks for the review  :)

**Comment by p0wn4j** [comment]
> That looks great. I've checked and the new version gets the same hits on LGTM as the old one. The other, more minor, comments from my previous review still stand.

I have copied ExecTainted.ql, ExecCommon.qll from java/ql/src/Security/CWE/CWE-078/ path.

**Comment by owen-mc** [comment]
https://lgtm.com/query/4250181641930060813/

**Pull request merged**